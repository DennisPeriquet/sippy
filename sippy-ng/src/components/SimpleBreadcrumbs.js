import { Link } from 'react-router-dom'
import Breadcrumbs from '@material-ui/core/Breadcrumbs'
import PropTypes from 'prop-types'
import React from 'react'
import Typography from '@material-ui/core/Typography'

export default function SimpleBreadcrumbs(props) {
  return (
    <Breadcrumbs aria-label="breadcrumb">
      <Link to={'/release/' + props.release}>Overview</Link>

      {props.previousPage ? props.previousPage : ''}

      {props.currentPage ? <Typography>{props.currentPage}</Typography> : ''}
    </Breadcrumbs>
  )
}

SimpleBreadcrumbs.propTypes = {
  release: PropTypes.string.isRequired,
  previousPage: PropTypes.element,
  currentPage: PropTypes.string,
}
