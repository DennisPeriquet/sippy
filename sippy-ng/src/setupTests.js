/** @jest-environment setup-polly-jest/jest-environment-node */

import '@testing-library/jest-dom'
import Enzyme from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import fetch from 'node-fetch'
import path from 'path'
import { setupPolly } from 'setup-polly-jest'
import toDiffableHtml from 'diffable-html'

// https://github.com/mui-org/material-ui/issues/21293
export const withoutMuiID = (wrapper) => toDiffableHtml(wrapper.html()
  .replace(/id="mui-[0-9]*"/g, '')
  .replace(/aria-labelledby="(mui-[0-9]* *)*"/g, ''))

Enzyme.configure({ adapter: new Adapter() })

// To stub out fetch(), we need to use node-fetch and this line:
global.fetch = fetch

// See: https://github.com/vuejs/vue-test-utils/issues/974
global.requestAnimationFrame = cb => cb()

// When injecting multiple filters, material table complains.
const originalError = console.error.bind(console.error)

global.console.error = (log) => {
  if (!log.toString().includes('Warning: Failed')) {
    originalError(log.toString())
  }
}

// Set API URL for Sippy
process.env.REACT_APP_API_URL = 'http://localhost:8080'

// Default PollyJS mode is replay, set to record to record
// new API calls.
if (!process.env.POLLY_MODE) {
  process.env.POLLY_MODE = 'replay'
}

export const setupDefaultPolly = () => {
  const context = setupPolly({
    mode: process.env.POLLY_MODE,
    adapters: [require('@pollyjs/adapter-node-http')],
    persister: require('@pollyjs/persister-fs'),
    persisterOptions: {
      fs: {
        recordingsDir: path.resolve(__dirname, '../__recordings__')
      }
    }
  })
  return context
}

export const expectLoadingPage = (wrapper) => expect(wrapper.find('p').contains('Loading...'))

jest.mock('react-chartjs-2', () => ({
  Bar: () => null,
  Chart: () => null,
  ChartComponent: () => null,
  Doughnut: () => null,
  Line: () => null,
  PolarArea: () => null
}))
