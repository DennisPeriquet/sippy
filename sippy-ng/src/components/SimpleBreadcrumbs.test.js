import { act } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import { mount } from 'enzyme'
import { withoutMuiID } from '../setupTests'
import React from 'react'
import SimpleBreadcrumbs from './SimpleBreadcrumbs'

jest.useRealTimers()

describe(SimpleBreadcrumbs, () => {
  it('shows overview', async () => {
    let wrapper
    await act(async () => {
      wrapper = mount(
        <BrowserRouter>
          <SimpleBreadcrumbs release="4.8" currentPage="Jobs" />
        </BrowserRouter>
      )
    })
    expect(wrapper.find('a[href="/release/4.8"]').exists()).toBe(true)
    expect(wrapper.find('p').text()).toContain('Jobs')
    expect(withoutMuiID(wrapper)).toMatchSnapshot()
  })

  it('shows current page', async () => {
    let wrapper
    await act(async () => {
      wrapper = mount(
        <BrowserRouter>
          <SimpleBreadcrumbs release="4.8" currentPage="Jobs" />
        </BrowserRouter>
      )
    })
    expect(wrapper.find('p').text()).toContain('Jobs')
    expect(withoutMuiID(wrapper)).toMatchSnapshot()
  })
})
