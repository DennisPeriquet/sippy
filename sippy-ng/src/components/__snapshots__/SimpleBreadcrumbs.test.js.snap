// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleBreadcrumbs shows current page 1`] = `
<nav class="MuiTypography-root MuiBreadcrumbs-root MuiTypography-body1 MuiTypography-colorTextSecondary"
     aria-label="breadcrumb"
>
  <ol class="MuiBreadcrumbs-ol">
    <li class="MuiBreadcrumbs-li">
      <a href="/release/4.8">
        Overview
      </a>
    </li>
    <li aria-hidden="true"
        class="MuiBreadcrumbs-separator"
    >
      /
    </li>
    <li class="MuiBreadcrumbs-li">
      <p class="MuiTypography-root MuiTypography-body1">
        Jobs
      </p>
    </li>
  </ol>
</nav>
`;

exports[`SimpleBreadcrumbs shows overview 1`] = `
<nav class="MuiTypography-root MuiBreadcrumbs-root MuiTypography-body1 MuiTypography-colorTextSecondary"
     aria-label="breadcrumb"
>
  <ol class="MuiBreadcrumbs-ol">
    <li class="MuiBreadcrumbs-li">
      <a href="/release/4.8">
        Overview
      </a>
    </li>
    <li aria-hidden="true"
        class="MuiBreadcrumbs-separator"
    >
      /
    </li>
    <li class="MuiBreadcrumbs-li">
      <p class="MuiTypography-root MuiTypography-body1">
        Jobs
      </p>
    </li>
  </ol>
</nav>
`;
